modules/misc.sh

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+# Copyright 2022 Northern.tech AS
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+# if nounset referencing empty array will cause an error. disabling it temporarily
+function check_and_merge_mender_configs() {
+    local nounset="false"
+
+    [ -o nounset ] && nounset="true"
+    set +o nounset
+    do_check_and_merge_mender_configs
+    [ "$nounset" == "true" ] && set -o nounset
+}
+
+function do_check_and_merge_mender_configs() {
+    local -a overlay_targets=()
+    local -r conf_path="work/rootfs/etc/mender/mender.conf"
+    local -r tmpfile="work/mergefiles/mender.conf"
+    local -r jqfilter="work/mergefiles/jqmergefilter"
+    local -r warn_message='The resulting image will not contain a Mender configuration file.
+To run ensure the file is there run a boostrap script and pass the output-dir as an overlay to mender-convert, 
+or supply your own overlay with a $OVERLAY/etc/mender/mender.conf file.'
+
+    if [ "${#overlays[@]}" -eq 0 ]; then
+        log_warn "No overlay specified. ${warn_message}"
+        return
+    fi
+
+    for overlay in "${overlays[@]}"; do
+        [ -f "${overlay}/etc/mender/mender.conf" ] && overlay_targets+=("${overlay}")
+    done
+
+    if [ "${#overlay_targets[@]}" -eq 0 ]; then
+        log_warn "Mender configuration file not found in any of the overlays. ${warn_message}"
+        return
+    fi
+
+    mkdir -p work/mergefiles
+    local -r menderconfs="${overlay_targets[@]/%//etc/mender/mender.conf}"
+
+    if [ "${#overlay_targets[@]}" -gt 1 ]; then
+        local message=""
+        message+="Attempting to merge /etc/mender/mender.conf in overlays. With respect to the overlays, the merge is done from top to bottom in the following order:"
+        message+="${overlay_targets[@]/#/$'\n'' * '}"
+        log_info "${message}"
+        cat << "EOF" > "${jqfilter}"
+        def deepmerge(a;b):
```

Comment:
Soo nice :heart_eyes: 

(Also glad this is not a one-liner :see_no_evil: )

Reply:
would have been cool tho

Reply:
:sweat_smile: